node-proxy/lib/utils/http-utils.js

```diff
@@ -29,12 +29,12 @@ exports.createProtocolServer = function(protocol, opts) {
       break;
     case 'https':
       var ssl_opts  = {
-        secureProtocol: 'SSLv23_method',
         secureOptions: constants.SSL_OP_NO_SSLv3
       };
       ssl_opts.ca   = fs.readFileSync(opts.ca);
       ssl_opts.cert = fs.readFileSync(opts.certificate);
       ssl_opts.key  = fs.readFileSync(opts.private_key);
+      ssl_opts.ciphers = opts.ciphers;
```

Comment:
It should be a config. With that said it seems like some error would not
hurt. However today we don't have this option and default ciphers are
chosen. So not having the option in the case of an upgrade should not be
too big of an issue.
On Jan 20, 2016 7:12 PM, "Miciah Dashiel Butler Masters" <
notifications@github.com> wrote:

> In node-proxy/lib/utils/http-utils.js
> <https://github.com/openshift/origin-server/pull/6349#discussion_r50342100>
> :
>
> >          secureOptions: constants.SSL_OP_NO_SSLv3
> >        };
> >        ssl_opts.ca   = fs.readFileSync(opts.ca);
> >        ssl_opts.cert = fs.readFileSync(opts.certificate);
> >        ssl_opts.key  = fs.readFileSync(opts.private_key);
> > +      ssl_opts.ciphers = opts.ciphers;
>
> My biggest concern is with how we handle upgrades.
>
> ...although I just checked and found that
> /etc/openshift/web-proxy-config.json is not marked %config(!), so that
> does not appear to be a problem.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin-server/pull/6349/files#r50342100>.
>